src/basic/process-util.c

```diff
@@ -2034,7 +2034,13 @@ int make_reaper_process(bool b) {
         return 0;
 }
 
-int posix_spawn_wrapper(const char *path, char *const *argv, char *const *envp, PidRef *ret_pidref) {
+int posix_spawn_wrapper(const char *path, char *const *argv, char *const *envp, const char *cgroup, PidRef *ret_pidref) {
+        /* Due to posix_spawn_attr_t cleanup we use a 'goto fail' pattern here, so make sure to declare all
+         * the _cleanup_ variables here at the top, otherwise they will not be initialized but the cleanup
+         * will be called regardless due to the jump on error. */
+        _unused_ _cleanup_close_ int cgroup_fd = -EBADF, pidfd = -EBADF;
+        _unused_ _cleanup_free_ char *resolved_cgroup = NULL;
```

Comment:
Hmm, it's OK to place cgroup_fd and resolved_cgroup in the if block? Only pidfd needs the special handling